File: corda-training-template/java-source/src/main/java/com/iou/java/state/IOUState.java
package com.iou.java.state;

import com.iou.java.contract.IOUContract;
import kotlin.collections.EmptyList;
import net.corda.core.contracts.ContractState;
import net.corda.core.crypto.CompositeKey;
import net.corda.core.crypto.Party;

import java.util.List;
import java.util.stream.Stream;

import static java.util.Collections.emptyList;
import static java.util.stream.Collectors.toList;

/**
 * The state object recording IOU agreements between two parties.
 *
 * A state must implement [ContractState] or one of its descendants.
 *
 *  * @param contract the contract which governs which transactions are valid for this state object.
 */
public class IOUState implements ContractState {
    private final IOUContract contract;

    public IOUState(IOUContract contract) {
        this.contract = contract;
    }

    @Override public IOUContract getContract() { return contract; }
    @Override public List<CompositeKey> getParticipants() {
        return emptyList();
    }
}

File: corda-training-template/kotlin-source/src/main/kotlin/com/iou/kotlin/state/IOUState.kt
package com.iou.kotlin.state

import com.iou.kotlin.contract.IOUContract
import net.corda.core.contracts.ContractState
import net.corda.core.contracts.LinearState
import net.corda.core.contracts.UniqueIdentifier
import net.corda.core.crypto.CompositeKey
import net.corda.core.crypto.Party
import java.security.PublicKey

/**
 * The state object recording IOU agreements between two parties.
 *
 * A state must implement [ContractState] or one of its descendants.
 *
 * @param contract the contract which governs which transactions are valid for this state object.
 */
data class IOUState(override val contract: IOUContract) : ContractState {

    override val participants: List<CompositeKey> get() = listOf()
}